### DISCUSSION TITLE
How do you get involved in development of friction?

### DISCUSSION THREAD

**Discussion by AlmostMikhail** [friction2d/friction#373] [Q&A] [answered]
I'm a unity c# dev with 2 years experience. What do I need to learn to get involved in friction 2d development?

**Comment by AlmostMikhail** [comment]
https://youtu.be/lEj_GfH8is8?si=je40jZSvVoGWRU6E

![44.jpg](https://github.com/user-attachments/assets/2f2f33b9-56d9-42c7-a372-043508f82882)


Here's an introduction to skia c++

**Comment by AlmostMikhail** [comment]
![Holy shit](https://github.com/user-attachments/assets/368b0c1c-de7b-4c88-aafb-d648fb5a6c40)
I don't know what I'm doing with this but I'm currently installing Qt and it going terribly. Remember not to install the commercial version of Qt. Because then you cant go back. I looked up on how to build Friction and Enve here(Since, on how to build is more clearer on Enve):
https://github.com/MaurycyLiebner/enve/blob/master/Source%20and%20building%20info.md
https://github.com/friction2d/friction/blob/main/BUILD.md
But, hopefully in a couple of hours Qt will be installed and I can mess around with the code but currently nobody offering help. I'll try to push through and update my progress here and if I don't then I quit.

**Comment by AlmostMikhail** [comment]
https://github.com/user-attachments/assets/e68b9a68-8a7f-4d83-a825-129a4048e3f2



I finally built Friction from source and I recorded the entire process so will linking video to that shortly.
It took me 2 days of work.
At first I thought you could just press play with the code in vscode but that didn't work so I looked more carefully at the instructions and ask @rodlie for help and chatgpt and eventually I did it. Now I just need to get into the coding part. 

Now do I use qt to edit software? Or is it all done in code and how do I preview my changes?

**Comment by rodlie** [comment, marked as answer]
> I'm a unity c# dev with 2 years experience. What do I need to learn to get involved in friction 2d development?

Hi.

Basic knowledge of C++ (and Qt) is needed to do anything.

I would suggest checking out some tutorials on C++/Qt before starting.

**Comment by kaixoo12** [comment]
@AlmostMikhail Hi, I'm creating a guide of how to contribute to friction. I know how to explain the code architecture but I forgot which issues I had when I was fi4st trying to build it. Could you write a tutorial to help beginners to compile friction, listing most common issues? That'd help a lot
